File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
    
    void recur(vector<char>& s,int index, int size,vector<char> & ans){
        if(index == size){
            return;
        }
        recur(s,index+1,size,ans);
        ans.push_back(s[index]);
    }
    
    void print(vector<char>& s,vector<char>& ans){
        int k = 0;
        for(auto it : ans){
            cout<<it<<" ";
            s[k++] = it;
        }
    }
    void reverseString(vector<char>& s) {
        vector<char> ans;
        recur(s,0,s.size(),ans);
        print(s,ans);
    }

};

File: 0344-reverse-string/0344-reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        recur(s,0,s.length-1);
    }
    
    public static char[] recur(char[] s,int start,int end){
        
        // base case
        if(start > end){
            return s;
        }
                
        //swapping character at start with character at end 
        char temp = s[start];
        s[start] = s[end];
        s[end] = temp;
        
        //recursion call
        recur(s,start + 1,end - 1);
        
        return s;
    }
}